server/container_create.go

```diff
@@ -1038,14 +1038,29 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 	}
 
 	// Get imageName and imageRef that are later requested in container status
-	status, err := s.StorageImageServer().ImageStatus(s.ImageContext(), images[0])
-	if err != nil {
-		return nil, err
+	var (
+		imgResult    *storage.ImageResult
+		imgResultErr error
+	)
+	if len(images) == 0 {
+		return nil, errors.New("cannot create a container when image is empty")
+	}
+	for _, img := range images {
+		imgResult, imgResultErr = s.StorageImageServer().ImageStatus(s.ImageContext(), img)
+		if err == nil {
+			break
+		}
+	}
+	if imgResultErr != nil {
+		return nil, imgResultErr
+	}
+	if imgResult == nil {
+		return nil, errors.New("ImageStatus() is really broken and should have returned an error")
```

Comment:
Looks like [`ImageStatus` is local too][1], and it never returns `nil` results without setting an error.  I'm fine dropping this guard on those grounds, although I'm also fine keeping it if maintainers feel the implementation might change and break that guarantee.

[1]: https://github.com/kubernetes-incubator/cri-o/blob/0b736bb43fb9f6ffa29b477be9e292bec3686ef4/pkg/storage/image.go#L322-L375

Reply:
Dropped the check. Since imgResult will be not be nil if the error is nil.